### ISSUE TITLE
AttributeError: type object 'RegularPolygon' has no attribute 'xy' with matplotlib version 3.4.1

### ISSUE THREAD

**Issue by 1kastner** [paulbrodersen/netgraph#23]
I just created a fresh install from this conda environment file:

```
name: myNewEnv
channels:
  - conda-forge
dependencies:
  - python=3.8
  - jupyterlab=3
  - ipympl
  - matplotlib
  - networkx
  - pip
  - pip:
    - netgraph
```

Then I wanted to just import the libraries, i.e.:

```
import networkx 
import netgraph
import matplotlib.pyplot as plt 
import matplotlib.image as mpimg
```

But here was was stopped with the following exception:

```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
~\Anaconda3\envs\myNewEnv\lib\site-packages\netgraph\_main.py in RegularPolygon()
    605     try: # seems deprecated for newer versions of matplotlib
--> 606         center = property(matplotlib.patches.RegularPolygon._get_xy,
    607                           matplotlib.patches.RegularPolygon._set_xy)

AttributeError: type object 'RegularPolygon' has no attribute '_get_xy'

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
<ipython-input-1-8690930b5b60> in <module>
      1 import networkx  
----> 2 import netgraph  
      3 import matplotlib.pyplot as plt 
      4 import matplotlib.image as mpimg 

~\Anaconda3\envs\myNewEnv\lib\site-packages\netgraph\__init__.py in <module>
     79 __email__ = "paulbrodersen+netgraph@gmail.com"
     80 
---> 81 from ._main import (draw,
     82                     draw_nodes,
     83                     draw_node_labels,

~\Anaconda3\envs\myNewEnv\lib\site-packages\netgraph\_main.py in <module>
    597 
    598 
--> 599 class RegularPolygon(matplotlib.patches.RegularPolygon):
    600     """
    601     The API for matplotlib.patches.Circle and matplotlib.patches.RegularPolygon in matplotlib differ substantially.

~\Anaconda3\envs\myNewEnv\lib\site-packages\netgraph\_main.py in RegularPolygon()
    607                           matplotlib.patches.RegularPolygon._set_xy)
    608     except AttributeError:
--> 609         center = matplotlib.patches.RegularPolygon.xy
    610 
    611 

AttributeError: type object 'RegularPolygon' has no attribute 'xy'
```

The execution of the code `import matplotlib; print(matplotlib.__version__)` results in `3.4.1`.

**Comment by shenzihu** [comment]
I have same problem. 'AttributeError: type object 'RegularPolygon' has no attribute '_get_xy''

**Comment by paulbrodersen** [comment]
Hi both, thank you for raising the issue (and confirming it). This is somewhat mysterious to me. The class that is throwing an error is the following: 

```python
class RegularPolygon(matplotlib.patches.RegularPolygon):
    """
    The API for matplotlib.patches.Circle and matplotlib.patches.RegularPolygon in matplotlib differ substantially.
    This class tries to bridge some of these gaps by translating Circle methods into RegularPolygon methods.
    """

    try: # seems deprecated for newer versions of matplotlib
        center = property(matplotlib.patches.RegularPolygon._get_xy,
                          matplotlib.patches.RegularPolygon._set_xy)
    except AttributeError:
        center = matplotlib.patches.RegularPolygon.xy
```

I last amended this class when matplotlib deprecated the `_get_xy` and `_set_xy` methods back in matplotlib version 2.x such that netgraph would remain compatible with previous matplotlib versions.  
According to the error trace, 

```shell
AttributeError: type object 'RegularPolygon' has no attribute 'xy'
```

the `xy` property seems to be missing as well now. 

However, if you look at the current source code in matplotlib stable, the attribute still exists and is used:

```python
class RegularPolygon(Patch):
    """A regular polygon patch."""

    def __str__(self):
        s = "RegularPolygon((%g, %g), %d, radius=%g, orientation=%g)"
        return s % (self.xy[0], self.xy[1], self.numvertices, self.radius,
                    self.orientation)

    @docstring.dedent_interpd
    def __init__(self, xy, numVertices, radius=5, orientation=0,
                 **kwargs):
        """
        Parameters
        ----------
        xy : (float, float)
            The center position.
        numVertices : int
            The number of vertices.
        radius : float
            The distance from the center to each of the vertices.
        orientation : float
            The polygon rotation angle (in radians).
        **kwargs
            `Patch` properties:
            %(Patch_kwdoc)s
        """
        self.xy = xy
        self.numvertices = numVertices
        self.orientation = orientation
        self.radius = radius
        self._path = Path.unit_regular_polygon(numVertices)
        self._patch_transform = transforms.Affine2D()
        super().__init__(**kwargs)

    def get_path(self):
        return self._path

    def get_patch_transform(self):
        return self._patch_transform.clear() \
            .scale(self.radius) \
            .rotate(self.orientation) \
            .translate(*self.xy)
```

Unfortunately, I don't have access to matplotlib 3.4.1 at the moment. Hence I can't test against that version myself. 
With version 3.3.4, everything seems fine on my end:

```
(matplotlib) 1 paul@paul-XPS-15-9560:~$ pip list                                                                                                                                
Package          Version                                                                                                                                                        
---------------- -------                                                                                                                                                        
backcall         0.2.0                                                                                                                                                          
cycler           0.10.0                                                                                                                                                         
decorator        5.0.7                                                                                                                                                          
ipython          7.7.0                                                                                                                                                          
ipython-genutils 0.2.0                                                                                                                                                          
jedi             0.18.0         
kiwisolver       1.3.1
line-profiler    3.2.1
matplotlib       3.3.4
memory-profiler  0.58.0
numpy            1.19.5
parso            0.8.2
pexpect          4.8.0
pickleshare      0.7.5
Pillow           8.2.0
pip              21.1
pkg-resources    0.0.0
prompt-toolkit   2.0.10
psutil           5.8.0
ptyprocess       0.7.0
Pygments         2.8.1
pyparsing        2.4.7
python-dateutil  2.8.1
setuptools       56.0.0
six              1.15.0
traitlets        4.3.3
wcwidth          0.2.5
wheel            0.36.2
(matplotlib) paul@paul-XPS-15-9560:~$ ipython
Python 3.6.9 (default, Jan 26 2021, 15:33:00) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.7.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import matplotlib

In [2]: %paste
class RegularPolygon(matplotlib.patches.RegularPolygon):
    """
    The API for matplotlib.patches.Circle and matplotlib.patches.RegularPolygon in matplotlib differ substantially.
    This class tries to bridge some of these gaps by translating Circle methods into RegularPolygon methods.
    """

    try: # seems deprecated for newer versions of matplotlib
        center = property(matplotlib.patches.RegularPolygon._get_xy,
                          matplotlib.patches.RegularPolygon._set_xy)
    except AttributeError:
        center = matplotlib.patches.RegularPolygon.xy
## -- End pasted text --

In [3]: 
```

**Would either of you humor me and drop their matplotlib version down to 3.3.4 or lower, and test if the issue persists?** That is the latest version available on pip, and hence the latest version that I can test against. I would like to confirm that this piece of code breaks due to a change in matplotlib between version 3.3.4 and version 3.4.1.

**Comment by shenzihu** [comment]
THX bro，the problem has been solved when the matplotlib version changed from 3.4.1 to 3.3.4。

**Comment by paulbrodersen** [comment]
Thanks for confirming that everything works fine with matplotlib version 3.3.4. 

I have a major update in the works after which netgraph will no longer use the RegularPolygon class for node artists. This should fully resolve the issue. In the meantime, I hope that using matplotlib version 3.3.4 is an acceptable workaround.

**Comment by paulbrodersen** [comment]
Just released a new major version (4.0.0), which passes all tests with matplotlib 3.4.2. The issue should now be fully resolved.

**Comment by 1kastner** [comment]
Cool, thank you!
